app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :username, presence: true, uniqueness: { case_sensitive: false },
+                       length: { minimum: 3, maximum: 25 }
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 50 },
+                    uniqueness: { case_sensitive: false },
+                    format: { with: VALID_EMAIL_REGEX }
+  PASSWORD_REQUIREMENTS = /\A
+    (?=.{8,})
+    (?=.*\d)
+    (?=.*[a-z])
+    (?=.*[A-Z])
+  /x.freeze
+
+  validates :password, format: PASSWORD_REQUIREMENTS
```

Comment:
- [ ] Kindly validate that the presence of the password is true.

Reply:
Thank you for the review and suggestion.

- I have validated the password.